v7/navstories/navstories.py

```diff
@@ -26,24 +26,146 @@
 
 from __future__ import unicode_literals
 from nikola.plugin_categories import ConfigPlugin
+from nikola import utils
+from nikola.utils import LOGGER
 
+import re
 
 class NavStories(ConfigPlugin):
     """Add all stories to the navigation bar."""
 
     name = 'navstories'
     dates = {}
 
+    conf_vars = ['TRANSLATIONS', 'NAVSTORIES_PATHS', 'NAVSTORIES_MAPPING', 'NAVIGATION_LINKS_POST_NAVSTORIES']
+
+    # Indention for each level deeper in a submenu, than the highest level in that submenu
+    # overwriiten by site.config['NAVSTORIES_SUBMENU_INDENTION'] if defined
+    navstories_submenu_indention = '* '
+
+    def map_to_menu(self, entries):
+        """
+        Map form list of pages going into menu to tuble of format as NAVIGATION_LINKS and NAVIGATION_LINKS_POST_NAVSTORIES
+        List format:
+        - List of "top level entry"
+          - List
+            - Top level Menu text, or None if auto-mapped, i.e. not in NAVSTORIES_MAPPING
+            - List of pages in the top menu entry
+              - List of:
+                - navpath: List containing navigation hieracy (permalink without langinfo (initial /en/) and without the NAVSTORIES_PATHS (e.g. /pages/))
+                - Permalink
+                - Page title
+        Example:
+        [   [   'Menu text for nav',
+                [   [['nav'], '/pages/nav/', 'nav/'],
+                    [['nav', 'p2', 'a'], '/pages/nav/p2/a/', 'Page 2a'],
+                    [['nav', 'P2', 'b'], '/pages/nav/P2/b/', 'Page 2b'],
+                    [['nav', 'P2', 'a'], '/pages/nav/P2/a/', 'Page 2a'],
+                    [['nav', 'p2'], '/pages/nav/p2/', 'Page 2'],
+                    [['nav', 'p1'], '/pages/nav/p1/', 'Side 1']]],
+            [   'Menu text for A',
+                [[['A', 'last'], '/pages/A/last/', 'Page title for A/last']]],
+            [   None,
+                [[['B', 'cde'], '/pages/B/cde/', 'Page title for B/cde']]]]
+        """
+        ret = []
+        for topent in entries:
+            # Determine toplevel menu name
+            # - If not None, use the name
+            # - If None, use title of page if page with navpath length=1 exist, else navpath[0]
+            title = topent[0]
+            if not title:
+                # Default is 1th level of navpath
+                title = topent[1][0][0][0]
```

Comment:
Why one, zero, zero, zero? Those indexes are really hard to follow.

That said, you could get rid of at least one by doing

```py
for title, topent_one in entries:
```

(come up with better name for `topent_one`), or perhaps OrderedDict (as noted below).